CONTRIBUTING.md

```diff
@@ -0,0 +1,32 @@
+# Contributing to YSwift
+
+This project is a Swift language layer over a binding to the Rust [Yrs](https://github.com/y-crdt/y-crdt) library using Mozilla's [UniFFI](https://github.com/mozilla/uniffi-rs/) project.
+
+## Issues and project work
+
+Issues for `YSwift` are tracked on GitHub at https://github.com/y-crdt/yswift/issues.
+We typically keep our roadmap plans within one of those issues, and use issues to loosely track out progress.
+
+This project keeps track of decision points and choices in the library development in a [Developer's Log](./devnotes/DevLog.md), and has a [release process documented](./devnotes/release-process.md).
+
+## Local Development Setup
+
+Scripts in the project build an XCFramework for iOS and macOS using UniFFI from the Rust sources as defined in [lib/Cargo.toml](./lib/Cargo.toml).
+The binary included in the released XCFramework is intimately tied to the code that UniFFI generates into `lib/swift/scaffold`.
+
+The rough pattern of dependencies that are reflected in `Package.swift`:
+
+```
+   +-----------+      +--------+      +--------+
+   | yniffiFFI | <--- | Yniffi | <--- | YSwift |
+   +-----------+      +--------+      +--------+
+ C static library       UniFFI      Developer created
+  from Rust lib        generated      Swift overlay
+```
+
+If you are working on any of the layers within the `lib` directory (the targets `yniffiFFI` or `Yniffi`) then set a local environment variable `LOCALDEV` to true and regenerate the XCFramework using the script `scripts/build-xcframework.sh`.
+The [Package.swift](./Package.swift) file is configured to look for the `LOCALDEV` environment variable and use a local reference to the XCFramework if it is set.
+Without that environment variable set, `Package.swift` uses the latest release version of the XCFramework.
```

Comment:
I think it'd be great to clarify how to do that correctly. For example:

If want to do local development then run this command in your terminal:
```
export LOCALDEV=true
```
If you want to cut release then run this command in your terminal:
```
export LOCALDEV=false
```
I am really bad at understanding the Unix parts of bash scripting, my (probably stupid) concern here is that couldn't it possibly affect or be affected by someone else setting LOCALDEV env variable in other places?

Reply:
In case if it could be affected from other places – what do you think if we just prefix the envrionment with `YSWIFT_`?

Reply:
The way the code operates, if there's any value in the `LOCALDEV` environment variable, it'll attempt to use the binaryTarget from a local reference, but fortunately it's fairly external to the release process itself, and more about having a default-immediately-usable-externally package here while still allowing you, me, Yuki-san, and any other contributors to be able to build our own XCFrameworks locally to iterate on the Rust level bindings when we need to.

Your point is great though - and I'll make it more explicit when you set it, and put in something about how to unset it and the effects.

And no issue adjusting the environment variable - `YSWIFT_LOCALDEV` it'll be!